File: Shells/Cpp/examples/BTSolverExample.cpp
#include"../cspSolver/BTSolver.cpp"
#include"../sudoku/SudokuBoardGenerator.hpp"
#include"../sudoku/SudokuFile.hpp"

#include<iostream>

int main(){
    SudokuFile sf = generateBoard(4,2,2,3);
    BTSolver solver = BTSolver(sf);
	std::cout<<"Below is a randomly-generated sudoku file."<<std::endl;
    std::cout<<solver.getSolution().toString()<<std::endl;

    solver.setConsistencyChecks("None");

    solver.setValueSelectionHeuristic("None");

    solver.setVariableSelectionHeuristic("None");

    
    solver.solve();
	std::cout<<"solve_complete"<<std::endl;

    if(solver.haveSolution()){
		std::cout<<"has solution"<<std::endl;
        solver.printSolverStats();
        std::cout<<solver.getSolution().toString()<<std::endl;
    }
    else
        std::cout<<"Failed to find a solution"<<std::endl;
    return 0;
}

File: Shells/Java/src/examples/BTSolverExample.java
package examples;

import cspSolver.BTSolver;
import cspSolver.BTSolver.ConsistencyCheck;
import cspSolver.BTSolver.ValueSelectionHeuristic;
import cspSolver.BTSolver.VariableSelectionHeuristic;
import sudoku.SudokuBoardGenerator;
import sudoku.SudokuFile;

public class BTSolverExample {

	public static void main(String[] args)
	{
		SudokuFile sf = SudokuBoardGenerator.generateBoard(9, 3, 3, 12);
		BTSolver solver = new BTSolver(sf);
		
		solver.setConsistencyChecks(ConsistencyCheck.None);
		solver.setValueSelectionHeuristic(ValueSelectionHeuristic.None);
		solver.setVariableSelectionHeuristic(VariableSelectionHeuristic.None);
		
		Thread t1 = new Thread(solver);
		try
		{
			t1.start();
			t1.join(60000);
			if(t1.isAlive())
			{
				t1.interrupt();
			}
		}catch(InterruptedException e)
		{
		}


		if(solver.hasSolution())
		{
			solver.printSolverStats();
			System.out.println(solver.getSolution());	
		}

		else
		{
			System.out.println("Failed to find a solution");
		}

	}
}
